components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/password/PasswordRecoveryManagerImpl.java

```diff
@@ -246,6 +320,50 @@ public SuccessfulPasswordResetDTO reset(String resetCode, char[] password, Map<S
         return buildSuccessfulPasswordUpdateDTO();
     }
 
+    /**
+     * Reset the password for password recovery, if the password reset code is valid.
+     *
+     * @param resetCode  Password reset code.
+     * @param password   New password.
+     * @param properties Properties.
+     * @return SuccessfulPasswordResetDTO {@link SuccessfulPasswordResetDTO} object which contain the information
+     * for a successful password update.
+     * @throws IdentityRecoveryException Error while resetting the password.
+     */
+    @Override
+    public SuccessfulPasswordResetDTO reset(String resetCode, String confirmationCode, char[] password, Map<String, String> properties)
+            throws IdentityRecoveryException {
+
+        // Validate the password.
+        if (ArrayUtils.isEmpty(password)) {
+            throw Utils.handleClientException(
+                    IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_NO_PASSWORD_IN_REQUEST.getCode(),
+                    IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_NO_PASSWORD_IN_REQUEST.getMessage(), null);
+        }
+        String newPassword = String.valueOf(password);
+        NotificationPasswordRecoveryManager notificationPasswordRecoveryManager = NotificationPasswordRecoveryManager
+                .getInstance();
+        Property[] metaProperties = buildPropertyList(null, properties);
+        try {
+            notificationPasswordRecoveryManager.updatePassword(resetCode, confirmationCode, newPassword, metaProperties);
+        } catch (IdentityRecoveryServerException e) {
+            String errorCode = Utils.prependOperationScenarioToErrorCode(e.getErrorCode(),
+                    IdentityRecoveryConstants.PASSWORD_RECOVERY_SCENARIO);
+            throw Utils.handleServerException(errorCode, e.getMessage(), null);
```

Comment:
```suggestion
            throw Utils.handleServerException(errorCode, e.getMessage(), null);
```
why the null is passed except for e for the throwable? 
The comment applies to every place where null is used